siae/daemon_test.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/NebulousLabs/Sia/build"
+)
+
+// TestStartDaemon probes the startDaemon function.
+func TestStartDaemon(t *testing.T) {
+	testDir := build.TempDir("siad", "TestStartDaemon")
+	config.Siad.NoBootstrap = false
+	config.Siad.APIaddr = "localhost:45170"
+	config.Siad.RPCaddr = ":45171"
+	config.Siad.HostAddr = ":45172"
+	config.Siad.SiaDir = testDir
```

Comment:
you should be able to use ":0", but you'll need to call various methods to discover which port the OS has assigned you. See other testing code for examples